src/subscription_manager/cli_command/register.py

```diff
@@ -387,6 +388,9 @@ def _prompt_for_environment(self):
         """
         By breaking this code out, we can write cleaner tests
         """
+        if not self._is_atty():
+            raise system_exit(os.EX_USAGE, "Error: --environments parameter not provided!")
```

Comment:
error messages shown to the user must be translatable; use the `gettext()` function, also usually available as `_()`